raster/r.horizon/main.c

```diff
@@ -499,6 +499,12 @@ int main(int argc, char *argv[])
         if (nbufferZone == 0.)
             nbufferZone = bufferZone;
 
+        /* adjust buffer to multiples of resolution */
+        ebufferZone = (int)(ebufferZone / stepx) * stepx;
+        wbufferZone = (int)(wbufferZone / stepx) * stepx;
+        sbufferZone = (int)(sbufferZone / stepy) * stepy;
+        nbufferZone = (int)(nbufferZone / stepy) * stepy;
+
```

Comment:
What's the wanted or expected behaviour when the added buffer is small, that would be rounded down to 0 cells more? Does specifying any buffer size bigger than 0 means that it should at least have another pixel? (I'm missing some context on how processing rasters work overall here).

I clearly understand the issue that is solved though, that a zone in a raster (aligned to a grid defined by the pixel's size), is shifted when increasing the zone studied, since it was equivalent to having the origin of the grid defined at the start of the increased zone, shifting the grid, and having partial (unaccounted) pixels at the opposite side. (I see this like an image processing problem).

Reply:
The behavior I chose was to round down, so for 10m resolution, if you specify bufferzone 5m, you end up with 0 bufferzone. It could be also done so that you end up with 1 pixel bufferzone (10 m). I don't think it matters that much, practically you would use bufferzones of at least several pixels, or tens of pixels.